### ISSUE TITLE
Improving the visibility of explicit maps in beatmap listing

### ISSUE THREAD

**Issue by Miimari** [ppy/osu-web#7372]
Currently maps that are marked with the explicit tag are initially hidden, and can only be shown by turning the explicit content filter off

due to this option being hidden many players are likely to not know of it's existence, which leads to players not being able to find their desired content
(ex  [this](https://osu.ppy.sh/community/forums/topics/1259758?start=7944281),  [this](https://www.youtube.com/watch?t=1659&v=6a0QfS_bd40&feature=youtu.be&ab_channel=TQRL) & [this](https://www.youtube.com/watch?t=3383&v=6a0QfS_bd40&feature=youtu.be&ab_channel=TQRL))

some ways that may improve how this filter works:
- **make the toggle setting easier to find** - moving this out of 'more search options' so more people will know of it's existence
- **initially start the setting on ‘show’ rather than ‘hide’** - this way if people don't want to see explicit content they can turn it off (could b used in conjunction with above method)
- **remove the filter entirely** - similarly to spotify, an explicit tag would be enough to warn a user about the content ( + popup warning)

**Comment by Ephemeralis** [comment]
Given that explicit has been clarified to apply (mostly) to audio content, taking the Spotify approach here and leaving it solely as a warning tag with the filter removed is a reasonable approach in my eyes.

I think there was some confusion during implementation about what explicit would be covering, and subsequent revisions to the policy that governs this stuff has made it very clear since.

**Comment by peppy** [comment]
Completely removing it? Not just changing the default value? I guess it doesn't have much meaning or use as a filter if it is defaulting to show...

**Comment by Darius-Wattimena** [comment]
What about doing something completely different? We could instead show a popup to the user the first time they open the beatmap listing page, asking if they want to see explicit content or not.

If we don't want anything like this then I agree that taking the Spotify approach would probably be a reasonable choice.

**Comment by TicClick** [comment]
unless I am missing something, it was added in #6575 to pessimize explicit content regardless of its nature, and showing these beatmaps by default kind of defeats this purpose and does not go in line with the spirit of [wiki:VCC](https://osu.ppy.sh/wiki/VCC) and [wiki:SCR](https://osu.ppy.sh/wiki/SCR).

giving the toggle more visibility (for example, showing a one-time popup on the beatmap listing, or moving it to the user settings area) sounds like a better idea and makes the listing conform to _explicit_ user choice

**Comment by TicClick** [comment]
side note: it may also be worth it to make this filter more granular by splitting it into "explicit visuals" and "explicit audio"

**Comment by peppy** [comment]
Against making things even more complicated than they already are. We won't be adding *more* toggles here.

**Comment by NoffyNoffeh** [comment]
> Given that explicit has been clarified to apply (mostly) to audio content, taking the Spotify approach here and leaving it solely as a warning tag with the filter removed is a reasonable approach in my eyes.
> 
> I think there was some confusion during implementation about what explicit would be covering, and subsequent revisions to the policy that governs this stuff has made it very clear since.

I really strongly support this one so long as it happens every time. This way we don't decrease visibility on these maps too drastically while still making it VERY clear what you're getting in to.

**Comment by SylasOsu** [comment]
any updates on this? it's been a few months

**Comment by peppy** [comment]
No, there are no updates.

**Comment by kittsun** [comment]
Bump, would really like to see something happen with this as it's making some really good maps + songs disappear for a good majority of the playerbase.

**Comment by BlastTheKidd** [comment]
> Completely removing it? Not just changing the default value? I guess it doesn't have much meaning or use as a filter if it is defaulting to show...

Changing the default value on this seems to be probably the best case scenario for this kind of topic considering the concerns that were raised in this original proposal, that way the actual filter/warning can still serve their initial purpose. 

I don't think it's reasonable to just ignore this while reducing visibility of "Official content", aside from this it does not match the standards for how widely popular music platforms handle explicit content already so it would only be fair if that power was given to the user.

I'm posting this here because I realize it's becoming more of an issue as time goes on.

**Comment by peppy** [comment]
I think the best way forward is to make it more visible that the filter is being applied. There should likely be a bar beneath the main filter control showing each enabled filter in a bubble like display, at very least.

**Comment by kittsun** [comment]
I agree, the reason why this was a worry in the first place was because the filter was so hidden away; if it can be moved from the More Search Options (+ maybe give a warning on first visit?), I feel that's a good approach.

**Comment by peppy** [comment]
Discussed with @nanaya that a first step here would be to always display the explicit content toggle, rather than having it hidden in the "more" content. This is a simple change to improve visibility while we figure how to address this better going forward.

@arflyte thinking forward, do you have any design changes in the pipeline for the search filter area? We need filters which are currently applied to be highlighted in some way. I was thinking having a pill bar at the bottom may be a good solution, but open to anything really. Example of what I think could work UX-wise:

https://user-images.githubusercontent.com/191335/155079764-0656c833-4622-4bca-b87b-0ddcfc1056b8.mp4

**Comment by atl128** [comment]
This can be closed?

**Comment by peppy** [comment]
No action has been taken yet. There's no reason to close it yet.

**Comment by jamesjan3** [comment]
would love to see this revisitied!

**Comment by Userlog2** [comment]
Maybe songs with Explicit marks are shown by default, with an option to "opt out" and hide them all in account settings instead of "opt in" (while still keeping the "Explicit Content" warning on the map's page itself); could work (except for vulgar titles).
